reservation/views.py

```diff
@@ -1,3 +1,40 @@
-from django.shortcuts import render
+from accommodation.models import Accommodation
+import json
 
-# Create your views here.
+from django.shortcuts   import render
+from django.http        import JsonResponse, HttpResponse
+from django.views       import View
+
+from user.utils         import login_decorator
+from reservation.models import Reservation, ReservationStatus
+from user.models        import User
+
+STATUS_BOOKED = 2
+
+class PurchaseView(View):
+    @login_decorator
+    def post(self, request):
+        try:
+            data    = json.loads(request.body)
+            user    = request.user
+
+            accommodation_id    = data['accommodation_id']
+            user_id             = user.id
+            start_date          = data['start_date']
+            end_date            = data['end_date']
+            total_price         = data['total_price']
+            total_guest         = data['total_guest']
+            status              = STATUS_BOOKED
+
+            Reservation.objects.create(
+                accommodation   = Accommodation.objects.get(id=accommodation_id),
+                user            = User.objects.get(id=user_id),
```

Comment:
```suggestion
                accommodation   = accommodation_id,
                user            = user_id,
```
두 id로도 지정이 가능해서 꼭 두 객체를 다시 SELECT 하지 않으셔도 좋습니다 :)
할당할 때 키 이름을 `accommodation`으로 해야할 지 `accommodation_id`해야할 지 테스트 해보셔야 할 것 같습니닷!